File: app/src/main/java/com/trofiventures/testchameleonframework/service/RetrofitClient.java
package com.trofiventures.testchameleonframework.service;

import com.facebook.stetho.okhttp3.StethoInterceptor;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.moshi.MoshiConverterFactory;

public class RetrofitClient {

    private Retrofit retrofit;

    public RetrofitClient() {
        init();
    }

    private void init() {
        OkHttpClient.Builder builder = new OkHttpClient.Builder();

        //builder.addInterceptor(new StethoInterceptor());
        HttpLoggingInterceptor httpLoggingInterceptor = new HttpLoggingInterceptor();
        httpLoggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);

        retrofit = new Retrofit.Builder()
                .baseUrl("http://34.211.194.70:8080/ca-1/")
                .addConverterFactory(MoshiConverterFactory.create())
                .client(builder.build())
                .build();
    }

    public ChameleonApi get() {
        if (retrofit != null)
            return retrofit.create(ChameleonApi.class);
        return null;
    }
}

File: chameleon/src/main/java/com/trofiventures/chameleon/service/RetrofitClient.kt
package com.trofiventures.chameleon.service

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.moshi.MoshiConverterFactory

class RetrofitClient {

    private var retrofit: Retrofit? = null

    init {
        init()
    }

    private fun init() {
        val builder = OkHttpClient.Builder()

        //builder.addInterceptor(new StethoInterceptor());
        val httpLoggingInterceptor = HttpLoggingInterceptor()
        httpLoggingInterceptor.level = HttpLoggingInterceptor.Level.BODY

        // todo change the url to production
        // baseUrl method
        retrofit = Retrofit.Builder()
                .baseUrl("http://mcpdev.mozido.com/corpay/ci/")
                .addConverterFactory(MoshiConverterFactory.create())
                .client(builder.build())
                .build()
    }

    fun get() = retrofit?.create(ChameleonApi::class.java)
}
